db/changes.go

```diff
@@ -774,9 +774,13 @@ func (db *Database) SimpleMultiChangesFeed(chans base.Set, options ChangesOption
 			}
 
 			if options.Revocations && db.user != nil {
-				channelsToRevoke := db.user.RevokedChannels(options.Since.SafeSequence())
-				for channel, triggeredBy := range channelsToRevoke {
-					feed := db.buildRevokedFeed(db.changeCache.getChannelCache().getSingleChannelCache(channel), options, triggeredBy, to)
+				revocationCheckSeq := options.Since.SafeSequence()
+				if options.Since.TriggeredBy > 0 {
+					revocationCheckSeq = options.Since.TriggeredBy - 1
+				}
+				channelsToRevoke := db.user.RevokedChannels(revocationCheckSeq)
```

Comment:
I'm not sure why the call to RevokedChannels should potentially use TriggeredBy - 1.  I thought it should just be using TriggeredBy, and the -1 is only needed below.

Reply:
`TestRevocationResumeSameRoleAndLowSeqCheck` shows the reason this is required:
- User has role1
- Role1 has ch1, ch2
- doc1 in ch1 at seq 10
- doc2 in ch2 at seq 11
- client does a pull. Last seq 11, has both docs.
- ch1 revoked from role at seq 20
- ch2 revoked from role at seq 30
At this point the invalSeq on the role is at 20 as we keep the oldest.
- Mutate doc1 seq 40
- Mutate doc2 seq 50

We start a revocation with say since 11. This will find that ch1 and ch2 are revoked and start sending, however, after sending the first the replication is interrupted and we have a since of 20:40.

If we just use '20' as triggered by it won't flag that ch1 and ch2 are revoked because `RevokedChannels` will say 'well at 20 you didn't have any channels so we're good'. So we need -1 for that.

Reply:
That makes sense, and explains some of the difficulty in naming revocationCheckSeq.  The name is a bit more appropriate when passed to user.RevokedChannels, but then less clear (at least to me) how it's being used when passed to buildRevokedFeed.

Reply:
Yeah the passed in value is then used for wasDocInChannelAtSeq for a similar reason